x-pack/packages/kbn-elastic-assistant/impl/assistant/api/prompts/use_fetch_prompts.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FindPromptsResponse } from '@kbn/elastic-assistant-common/impl/schemas/prompts/find_prompts_route.gen';
+import { useQuery } from '@tanstack/react-query';
+import { API_VERSIONS, ELASTIC_AI_ASSISTANT_PROMPTS_URL_FIND } from '@kbn/elastic-assistant-common';
+import { HttpSetup, IToasts } from '@kbn/core/public';
+import { i18n } from '@kbn/i18n';
+import { useAssistantContext } from '../../../assistant_context';
+
+export interface UseFetchPromptsParams {
+  signal?: AbortSignal | undefined;
+  consumer?: string;
+}
+
+/**
+ * API call for fetching prompts for current spaceId
+ *
+ * @param {Object} options - The options object.
+ * @param {string} options.consumer - prompt consumer
+ * @param {AbortSignal} [options.signal] - AbortSignal
+ *
+ * @returns {useQuery} hook for getting the status of the prompts
+ */
+
+export const useFetchPrompts = (payload?: UseFetchPromptsParams) => {
+  const {
+    assistantAvailability: { isAssistantEnabled },
+    http,
+  } = useAssistantContext();
+
+  const QUERY = {
+    page: 1,
+    per_page: 1000, // Continue use in-memory paging till the new design will be ready
+    filter: `consumer:${payload?.consumer ?? '*'}`,
+  };
+
+  const CACHING_KEYS = [
+    ELASTIC_AI_ASSISTANT_PROMPTS_URL_FIND,
+    QUERY.page,
+    QUERY.per_page,
+    QUERY.filter,
+    API_VERSIONS.internal.v1,
+  ];
+
+  return useQuery<FindPromptsResponse, unknown, FindPromptsResponse>(
+    CACHING_KEYS,
+    async () =>
+      http.fetch(ELASTIC_AI_ASSISTANT_PROMPTS_URL_FIND, {
+        method: 'GET',
+        version: API_VERSIONS.internal.v1,
+        query: QUERY,
```

Comment:
The long term goal for prompt settings table would be moving away from the in-memory table and control the pagination. We might want the `query` to be passed from the `payload` in the long run.

At the same time, is it possible to add the include the conversations that are using this prompt? The current implementation for fetching conversations that using this prompt is, it actually filtering `all` the conversations by their defaultSystemPrompt Id on the client side. This might not be great for performance when there are more and more conversations in the future.